### PR TITLE
Add cross-compilation to symbol mangling.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

BoringSSL potentially defines different symbols on different
architecture and platform combinations. This means that we've
historically missed mangling some symbol names when our script only
involved building on macOS.

We should build on more platforms to test this.

Modifications:

- Added support for cross compiling to Linux amd64, Linux aarch64, and
    Linux armv7 in symbol mangling script.
- Added support for compiling for iOS armv6 and arm64 in symbol mangling
    script.

Result:

Harder to run the mangling script, better results when you do.

### CODE REVIEW

File: scripts/vendor-boringssl.sh [hunk 1]
```diff
@@ -69,6 +71,67 @@ function namespace_inlines {
...
+
+        # Use this as the input to the mangle.
+        go run "${SRCROOT}/util/make_prefix_headers.go" -out "${HERE}/${DSTROOT}/include/openssl" "${TMPDIR}/symbols.txt"
+
+        # Remove the product, as we no longer need it.
+        $sed -i -e 's/MANGLE_START\*\//MANGLE_START/' -e 's/\/\*MANGLE_END/MANGLE_END/' "${HERE}/Package.swift"
+    )
+
+    # Now remove any weird symbols that got in and would emit warnings.
+    gsed -i -e '/#define .*\..*/d' "${DSTROOT}"/include/openssl/boringssl_prefix_symbols*.h
```
Comment (glbrntt) [inline scripts/vendor-boringssl.sh]: `$sed`? (Although I guess this makes no difference as we need Darwin anyway)

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
@weissi Any idea what's going on the api-breakage script?

**Comment by weissi** [comment]
@Lukasa the error is whilst compiling 

```
15:21:49 [352/354] Compiling a_bitstr.c
15:21:52 [353/372] Compiling NIOSSL LinuxCABundle.swift
15:21:52 <unknown>:0: warning: missing submodule 'CNIOBoringSSLShims'
15:21:52 /tmp/.check-api_rXeODo/repo/Sources/NIOSSL/SSLCertificate.swift:17:29: error: no such module 'CNIOBoringSSLShims'
15:21:52 @_implementationOnly import CNIOBoringSSLShims
15:21:52                             ^
```

**Comment by weissi** [comment]
this looks like missing dependencies?

**Comment by Lukasa** [comment]
Nope, it's part of the swift package. It's sitting right there.

**Comment by tomerd** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
Dang, the tests got beaten by the timeout by about 20s because they had to rebuild the images. Kicking again.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

File: scripts/vendor-boringssl.sh (hunk 1 above)
Comment (Lukasa) [reply scripts/vendor-boringssl.sh]: Good catch, fixed.

**Comment by tomerd** [comment]
@swift-nio-bot test this please

**Pull request merged**